admin/capabilities/class-capability-members-integration.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * @package WPSEO\Admin\Capabilities
+ */
+
+/**
+ * Integrates Yoast SEO capabilities with Members and other plugins that use the same filter.
+ */
+class WPSEO_Capability_Members_Integration {
+	/**
+	 * Registers the hooks.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		add_filter( 'members_get_capabilities', array( $this, 'get_capabilities' ) );
+		add_action( 'members_register_cap_groups', array( $this, 'action_members_register_cap_group' ) );
+	}
+
+	/**
+	 * Get the Yoast SEO capabilities.
+	 * Optionally append them to an existing array.
+	 *
+	 * @param  array $caps Optional existing capability list.
+	 * @return array
+	 */
+	public function get_capabilities( $caps = array() ) {
+		if ( ! did_action( 'wpseo_register_capabilities' ) ) {
+			do_action( 'wpseo_register_capabilities' );
```

Comment:
This is not required, as the capabilities should only be registered on activation or upgrade routines.
The registration registers the configured capabilities to the system.
They can just be retrieved by using the Capability Manager.

Reply:
This is only when there are roles that contain the capabilities.
For example: if you delete the SEO roles added by Yoast the added capabilities disappear from role manager plugins.
Most plugins just have a simple function that returns an array with all the plugin related capabilities but I chose to use your register functionality so you won't need to maintain the list at at multiple locations.

Reply:
Also, this function is only run if the `members_get_capabilities` filter is called.